test/parallel/market.rpc.check.test.ts

```diff
@@ -0,0 +1,473 @@
+
+      expect(sellPriceBefore).bnEqual(sellPriceWithImpact[0]);
+      expect(diff).bnLt(sellPriceAfter.muln(3).divn(1000));
+    });
+
+    test("Function calculateBuyPriceByMarket works correctly", async () => {
+      const buyPrice = await calculateBuyPriceByMarket(
+        liqId,
+        secondCurrency,
+        threshold.muln(2),
+      );
+
+      if (poolType === "Xyk") {
+        const poolBalance = await getBalanceOfPool(
+          firstCurrency,
+          secondCurrency,
+        );
+        priceCalculationRetro = await calculate_buy_price_rpc(
+          poolBalance[0],
+          poolBalance[1],
+          threshold.muln(2),
+        );
+      } else {
+        priceCalculationRetro = buyPrice;
+      }
+
+      const buyPriceReverse = await calculateBuyPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold.muln(2),
+      );
+
+      const tokenAmountBefore = await getTokensAccountInfo(
+        testUser.keyRingPair.address,
+        firstCurrency,
+      );
+
+      await signTx(
+        api,
+        Market.buyAsset(
+          liqId,
+          firstCurrency,
+          secondCurrency,
+          threshold.muln(2),
+        ),
+        testUser.keyRingPair,
+      ).then((result) => {
+        const eventResponse = getEventResultFromMangataTx(result);
+        expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      const tokenAmountAfter = await getTokensAccountInfo(
+        testUser.keyRingPair.address,
+        firstCurrency,
+      );
+
+      expect(buyPrice).bnEqual(priceCalculationRetro);
+      expect(buyPrice).not.bnEqual(buyPriceReverse);
+      expect(
+        stringToBN(tokenAmountBefore.free).sub(
+          stringToBN(tokenAmountAfter.free),
+        ),
+      ).bnEqual(buyPrice);
+    });
+
+    test("Function calculateBuyPriceWithImpact makes reliable calculations", async () => {
+      const buyPriceBefore = await calculateBuyPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold.muln(2),
+      );
+
+      const buyPriceWithImpact = await calculateBuyPriceWithImpact(
+        liqId,
+        firstCurrency,
+        threshold.muln(2),
+      );
+
+      await signTx(
+        api,
+        Market.buyAsset(
+          liqId,
+          firstCurrency,
+          secondCurrency,
+          threshold.muln(2),
+        ),
+        testUser.keyRingPair,
+      ).then((result) => {
+        const eventResponse = getEventResultFromMangataTx(result);
+        expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      const buyPriceAfter = await calculateBuyPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold.muln(2),
+      );
+
+      const diff = buyPriceAfter.sub(buyPriceWithImpact[1]);
+
+      expect(buyPriceBefore).bnEqual(buyPriceWithImpact[0]);
+      expect(diff).bnLt(buyPriceAfter.muln(3).divn(1000));
+    });
+  },
+);
+
+test("check getBurnAmount", async () => {
```

Comment:
rename to : Market - rpc - getBurnAmount  test

Reply:
Done